core/src/main/java/tc/oc/pgm/action/actions/WeatherAction.java

```diff
@@ -0,0 +1,40 @@
+package tc.oc.pgm.action.actions;
+
+import tc.oc.pgm.action.WeatherType;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.modules.WeatherMatchModule;
+
+public class WeatherAction extends AbstractAction<Match> {
+
+  private final Enum state;
+
+  public WeatherAction(Enum state) {
+    super(Match.class);
+    this.state = state;
+  }
+
+  public void trigger(Match match) {
+    WeatherMatchModule weatherMatchModule = match.getModule(WeatherMatchModule.class);
+
+    weatherMatchModule.allowChange();
+
+    switch (state) {
+      case WeatherType.CLEAR:
+        match.getWorld().setStorm(false);
```

Comment:
extract `match.getWorld()` outside of the switch, it's being called a ton of times